internal/jobservice/eventstojobs/eventstojobs.go

```diff
@@ -0,0 +1,107 @@
+package eventstojobs
+
+import (
+	"context"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/G-Research/armada/internal/jobservice/configuration"
+	"github.com/G-Research/armada/internal/jobservice/repository"
+	"github.com/G-Research/armada/pkg/api"
+	"github.com/G-Research/armada/pkg/client"
+)
+
+// Service that subscribes to events and stores JobStatus in the repository.
+type EventsToJobService struct {
+	queue                string
+	jobsetid             string
+	jobid                string
+	jobServiceConfig     *configuration.JobServiceConfiguration
+	jobServiceRepository repository.JobServiceRepository
+}
+
+func NewEventsToJobService(
+	queue string,
+	jobsetid string,
+	jobid string,
+	jobServiceConfig *configuration.JobServiceConfiguration,
+	jobServiceRepository repository.JobServiceRepository) *EventsToJobService {
+	return &EventsToJobService{
+		queue:                queue,
+		jobsetid:             jobsetid,
+		jobid:                jobid,
+		jobServiceConfig:     jobServiceConfig,
+		jobServiceRepository: jobServiceRepository,
+	}
+}
+
+// Subscribes to a JobSet from jobsetid
+func (eventToJobService *EventsToJobService) SubscribeToJobSetId(context context.Context) error {
+
+	return eventToJobService.streamCommon(&eventToJobService.jobServiceConfig.ApiConnection, context)
+}
+
+func (eventToJobService *EventsToJobService) streamCommon(clientConnect *client.ApiConnectionDetails, ctx context.Context) error {
+	var fromMessageId string
+	conn, connErr := client.CreateApiConnection(clientConnect)
+	eventToJobService.jobServiceRepository.SubscribeJobSet(eventToJobService.jobsetid)
+	if connErr != nil {
```

Comment:
if `connErr` is not nil and you couldn't create a connection is there anything useful you can do here?  I'm pretty sure all the goroutines below will fail in that case so you'd be much better just bailing here without creating them

Reply:
Yea you are right.

Reply:
I returned an error here.  I think we probably shouldn't panic?